### ISSUE TITLE
[RFC] Create a shared space to organize and track PDDE cross-team work

### ISSUE THREAD

**Issue by dblandin** [artsy/README#233]
## Proposal

Create a shared space to organize and track PDDE cross-team work.

## Reasoning

Tasks within the Product, Design, Data, and Engineering (PDDE) organization are often organized and tracked within individual product teams. Product teams within the organization have largely standardized on using Jira Software.

Outside of product teams, there are other structures in which colleagues collaborate on important cross-cutting efforts. So far we have [Practices](https://github.com/artsy/README/blob/0b35ba2c867b46f38c7215a89bce14a14f17a250/practices/README.md), Working Groups, and Task Forces. These different groups organize and track their tasks in different places.

**Examples**

- The Platform Practice group organizes its tasks within the Platform product team’s Jira Software project
- The On-Call Working group organizes its tasks within a kanban board in Notion
- The Design Systems group organizes its tasks within the open-source artsy/palette GitHub repository

The reduced visibility of work within these cross-team groups may have an adverse on planning within individual product teams. It can be difficult to anticipate how work managed outside of product team Jira Software projects may affect a team’s bandwidth in current or future sprints.

A solution might:

- Increase the visibility of goals/efforts
- Improve consistency to organizing/tracking efforts
- Enable improved collaboration between groups
- Allow for representation of tasks within product team sprint planning

## Exceptions

None yet

## Additional Context

I originally chatted about this casually with @iskounen. I've since brought it up in a couple different Practice / Working Group meetings and during the last Front-End Practice meeting.

## Open Questions

- Is this a problem felt by others?
- If so, what are some potential solutions?

## How is this RFC advanced?

Validate the problem and update the RFC with a specific proposal

## How is this RFC resolved?

- Create (or reuse) a Jira project to track non-product initiatives.
- Document recommendations for creating, organizing, and assigning tickets in artsy/README. 

Participation in this process will be opt-in for Practice, Working Group, and Task Force leaders. Much of the process for the new Jira project will be sourced from the discussions here but there will be another PR with its documentation to provide feedback on as well.

**Comment by sweir27** [comment]
I think I suggested this in a different meeting, but it would be extra cool if we could keep this all in Jira (and easily pull items into various product team sprints).

One way to share linked items across product teams is to organize them under an `Epic`, either on separate project boards (one for each working group/task force/etc.) or a catch-all project.

**Comment by joeyAghion** [comment]
I definitely feel this in a few places. The security working group is building its backlog in Jira for consistency, but there's still the risk of that work being "hidden" away from product teams.

There are at least 3 kinds of non-product-team forums today: practices, working groups, and task forces. The first 2 anticipate being long-lived while task forces collect around a specific problem and [I hope] can disband once solved or decided. All 3 can generate work, some of the discussion/research/experimentation variety, (which I think is fine to keep casual), and sometimes of the serious investment variety (which is critical to manage and recognize in the open).

It would be convenient to simply say that any significant work should be advocated for within existing teams and prioritized or planned as upkeep as usual. This makes the work visible, but also ensures it's aligned with other product work in terms of sequencing, timing, resourcing, etc. (They can even be distinguished by labels or epics if helpful.)

This appeals to me because it's not creating anything new and doesn't require individuals to monitor separate backlogs. It's a challenge though, because it relies on engineers advocating for work with cross-team impact and product managers understanding and prioritizing those efforts.

There are some alternatives worth considering, but I don't necessarily see them as viable solutions:
* In some cases I can imagine dedicated teams forming around the topics (e.g., security). In that case, a separate backlog makes total sense.
* We could choose to protect and dedicate some engineering capacity to these efforts, such as by saying that certain individuals are only 50% staffed to product teams, or all engineers have 20% protected for that purpose. I'd worry that this is out of alignment with our focus on business impact.

**Comment by SamRozen** [comment]
![36vb0p](https://user-images.githubusercontent.com/5985818/62159134-043bf480-b2df-11e9-83ce-018962bdec82.jpg)

**Comment by dblandin** [comment]
@williardx @GuillaumeDelg @joanna20 @junybrum :wave: 

I would love your input on this RFC! Particularly, as product managers, is this a problem you recognize? If so, have your teams experimented with any solutions?

**Comment by joanna20** [comment]
@dblandin Definitely always in favor of us keeping more things open and following the same simple processes! But I'm not sure I totally understand the pain points we're trying to solve for here, can you expand on those examples provided up top to explain the negative impact felt?

In the Galleries team if people are doing work that's part of a working group (outside the product team's scope) we've said they should create a ticket to track it alongside what the team committed to for the sprint if it helps them. Holistically that might not be the best solution because it won't link back to the original task, but I don't think this has bubbled up as a larger issue.

**Comment by dblandin** [comment]
> Definitely always in favor of us keeping more things open and following the same simple processes! But I'm not sure I totally understand the pain points we're trying to solve for here, can you expand on those examples provided up top to explain the negative impact felt?

@joanna20 For sure! I'll do my best to offer a bit more context.

## Context

We already have collective buy-in that the work taking place within Practices, Working Groups, and Task Forces is important but I don't believe that these tasks are often represented within product teams whether during sprint planning, standups, or at other times.

> In the Galleries team if people are doing work that's part of a working group (outside the product team's scope) we've said they should create a ticket to track it alongside what the team committed to for the sprint if it helps them.

I'm really happy to hear this! This is exactly the type of process I'd like to standardize on across the org.

Another challenge I've felt is that members of these groups don't have a clear process to follow that includes their product team when agreeing to take on work, regardless of the deadline. This can cause a stall in progress within groups if there's important sprint work ongoing and not enough bandwidth.

## Implementation

Here's one possible implementation utilizing Jira:

**Prep**

- We create a new Jira software project to track all non-product work.
- We create epics in the new project, aligning with current Practices, Working Groups, and Task Forces.

**Process**

- During a Practice/WG/TF meeting, potential action items are discussed and documented in notes.
- Recommended action items are ticketed in the shared Jira project.
- Practice/WG/TF members are responsible for advocating for tasks within their product team. Once accepted by a product team, the ticket remains in the original epic, but is transferred to the product team's Jira project so that it be included within their backlog and sprint.

I think this alleviates a few pain points by:

- Consolidating tasks across groups into one place
- Encouraging proactive communication of tasks within product teams
- Ensuring that members have the appropriate bandwidth to take on a task or reassign
- Better visibility on who's doing what for Practice/WG/TF organizers

**Comment by samchieng** [comment]
@dblandin @joanna20 @GuillaumeDelg @williardx 

We've been talking about this for our Palette process to make sure we can help allocate time during sprints to build out components. I think it could be helpful for 1) keeping everyone in PDDE updated about the current build status of different Palette components and 2) for helping record and prioritize Palette tickets with other sprint work. 

Previously it's been hard to keep track of Palette components being built (for design and so other teams can use them) in a central location, and thus advocating for them in grooming and planning. We've tried tracking this in a Github board, but because it's not in Jira the work doesn't get the same visibility.

Wondering if we could test this idea with just Palette to start? We're hoping to refine our process over time so it gets easier to maintain our design system, but I think this might help streamline our process (from design -> build -> QA & documentation).

**Comment by joeyAghion** [comment]
I'm +1 about most of these things. Especially:

> Practice/WG/TF members are responsible for advocating for tasks within their product team

> Encouraging proactive communication of tasks within product teams

These are just good practices that help teams prioritize their own upkeep work as well as relevant needs that emerge from practices, task forces, or working groups. They also get buy-in from product managers early and recognize that we expect *everyone* to play an active role and not just TLs or PMs.

I'm not 100% convinced about this:

> Consolidating tasks across groups into one place

Would we ever actually refer to the complete view of non-product tasks across teams? Another consideration is that different forums have different workflows or stages (columns). E.g., the security working group might use stages relating to triaging bounty submissions or reported vulnerabilities. Also, I don't currently know how a group could consistently sort or prioritize the issues within an epic, which would be an important workflow. (It's possible to sort "ready" items in the epic-filtered backlog view, but that doesn't sync exactly to the epic's own view.)

**Comment by iskounen** [comment]
@joeyAghion: regarding your question about referring to a complete view of non-product tasks, I think that we can get more value out of the epics if we break them out by initiative rather than by group (practice, workgroup, task force).

For example, there is an initiative to improve Volt build times, which both the Platform practice and the Galleries team are interested in. I would like to see an epic named "Reduce Volt Build Time" that anyone can add tickets to, and anyone can bring into a team sprint. This will allow stakeholders from across PDDE to see what's been done, what tasks have been proposed (as incomplete tickets), who has been active in that initiative, etc.

There is also a way to view the progress for each epic, which could be interesting if we start associating epics with OKRs. For example, if we set "Reduce Volt Build Time by 25%" as a key result for some objective, we could use this view to get an overview of how our progress.

<img width="258" alt="epics" src="https://user-images.githubusercontent.com/44589599/62890927-816d5d80-bd12-11e9-93b0-1c89cdf4efb0.png">

**Comment by iskounen** [comment]
@joeyAghion: regarding ongoing initiatives like triaging bug bounty notifications, we have [an epic like this in the Galleries project](https://artsyproduct.atlassian.net/browse/GALL-611) that keeps getting longer and longer. It's useful to have 1 place where we can find the solutions to previous issues, but eventually I would like to organize those to be easier to find using labels.

**Comment by joeyAghion** [comment]
Overall I like the vision you describe @iskounen. [That epic epic](https://artsyproduct.atlassian.net/browse/GALL-611) makes me think we're still fighting Jira a little to make it work for long-term work streams like bounties. Other than it being really long, do you find that it works well?

Essentially what I think we're saying is that non-product initiatives will have a new place to be surfaced, undergo discussion, and be prioritized, but then they'll need adopting by product teams and migrate to those projects. Correct?

**Comment by iskounen** [comment]
> Overall I like the vision you describe @iskounen. [That epic epic](https://artsyproduct.atlassian.net/browse/GALL-611) makes me think we're still fighting Jira a little to make it work for long-term work streams like bounties. Other than it being really long, do you find that it works well?

That epic epic has worked well because we get about 5 new issues each sprint, and we can search that epic for keywords like "Change Contract Date" to [find steps taken in the past](https://artsyproduct.atlassian.net/browse/GALL-2000) to address similar issues. As you mentioned, a lot of this value comes from us finding new ways to work with JIRA, and relies on our gallery liaisons diligently adding tags (i.e. [Change Contract Date]) to ticket titles, or engineers saving the rails console commands they ran in comments.

For the bug bounty program, I could see us getting some value out of translating our existing Trello tickets to JIRA tickets and categorizing them by vulnerability type, service, etc. to make triaging and assignging new incidents easier.

> Essentially what I think we're saying is that non-product initiatives will have a new place to be surfaced, undergo discussion, and be prioritized, but then they'll need adopting by product teams and migrate to those projects. Correct?

Correct. The step to migrate tickets to another project is as simple as moving a GALL ticket to the PLATFORM project (for example).

![move](https://user-images.githubusercontent.com/44589599/63021250-2badc800-be6e-11e9-93ec-2a206e3d9031.png)

**Comment by joeyAghion** [comment]
For special cases, it'll always be an option to create dedicated Jira projects. The principle should be the same though: as much as possible, work must be adopted by teams once it becomes significant or gets past the research/decision phase.

I'll just point out that the ["Tech updates / Dev omprovements" Jira project](https://artsyproduct.atlassian.net/projects/TUDI/issues) appears to have been created for a similar purpose (if not so organized). Maybe we can take it over and rename it to "Engineering," or just retire it in favor of a new project.

**Comment by iskounen** [comment]
+1 to starting out with the TUDI project. In the future, I'd be interested to see what the effects of each practice having their own project type in JIRA would be. I think that it could help "establish" practices (which are mostly an Enginering thing) as a bona fide component of the PDDE org.

**Comment by dblandin** [comment]
Thanks everyone for the engaging discussion! :100: 

I'm going to update the PR description with the following "How is this RFC resolved?" content:

---

- Create (or reuse) a Jira project to track non-product initiatives.
- Document recommendations for creating, organizing, and assigning tickets in artsy/README. 

Participation in this process will be opt-in for Practice, Working Group, and Task Force leaders. Much of the process for the new Jira project will be sourced from the discussions here but there will be another PR with its documentation to provide feedback on as well.

---

@iskounen If you're interested and have time, I'd love to pair on the Jira software configuration with you.

**Comment by anandaroop** [comment]
I'm 👍  on the RFC.

One more data point: as it happens, the time zone working group is organized around a Jira epic that lives on the Platform project — https://artsyproduct.atlassian.net/browse/PLATFORM-1369.

Actual work has been surfaced during Grow sprint planning and some tickets have been created in the Grow backlog, while still being attached to this Platform epic.

**Comment by mbilokonsky** [comment]
I see this is still open and I just want to say I think it's broadly speaking a really great idea. When I have an upkeep day, or if I've finished all of my tickets for a sprint, it would be tremendously powerful for me to be able to go to a general PDDE backlog arranged by initiative and pull a ticket or two if it's something I feel like I can contribute to. Centralizing this stuff allows cross-team discovery etc, and we can even organize mob days where we attack the initiative backlog together as a company etc.

I really like this.

**Comment by dblandin** [comment]
Sorry for the delayed update here. This is quite a bit to document.

@iskounen and I looked at options in Jira and determined that trying to put everything into one project would work against how Jira usually operates and that it would be more useful to document the    - Enable the Epics panel creation of a new project for any Task Force / Working Group / Practice that wishes to adopt it.

So here is some proposed documentation for such projects. We might be able to significantly decrease the amount of work involved by setting up a template project and allowing new projects to copy the configuration from that, which is available during the creation process. But to document everything, here's the process starting from scratch.

A next step might be to move this documentation into README. I'm hoping that at least the On-Call Working Group and Design Systems can start to use these projects soon.

### Instructions

1. **Create a new Jira Project**

    You'll be prompted to choose a Project Type between Classic and Next-gen.

    Choose Classic which offers more flexibility.

    ![screenshot 1568926289](https://user-images.githubusercontent.com/123595/65280137-bb920200-dafd-11e9-862f-bd2ddf96c054.png)

    - Keep the default Kanban template
    - Fill in the name of your project (e.g. On-Call Working Group)

    ![screenshot 1568926455](https://user-images.githubusercontent.com/123595/65280461-65718e80-dafe-11e9-84b5-5cb585505677.png)

2. **Update the Project Settings**

    - Click into the project settings

    ![screenshot 1568926629](https://user-images.githubusercontent.com/123595/65280674-ea5ca800-dafe-11e9-9879-f952de1aee50.png)

    - Update the project lead, description, and URL of the project

    ![screenshot 1568931225](https://user-images.githubusercontent.com/123595/65284998-3c0a3000-db09-11e9-8496-986ead557139.png)

3. **Associate the project's workflow scheme with the shared "Product Workflow Scheme"**

    This allows us to easily move the tickets into other projects later.

    - Navigate from Project Settings to Workflows and click "Switch Scheme".

    ![screenshot 1568932058](https://user-images.githubusercontent.com/123595/65286179-9a84dd80-db0c-11e9-998b-eca49e76efe5.png)

    - Select "Product Workflow Scheme" from the dropdown menu and click "Associate"
 
    ![screenshot 1568932095](https://user-images.githubusercontent.com/123595/65286181-9ce73780-db0c-11e9-95e2-a03eb4bfdf94.png)

4. **Update the board settings to enable the backlog**

    - On either the backlog or board page, click the ... button and click Board Settings
    - Configure the columns as shown below, mapping issue statuses to their appropriate columns
    - Enable the Epics panel

    ![screenshot 1568932641](https://user-images.githubusercontent.com/123595/65286156-8e008500-db0c-11e9-8357-06a3d8a0586d.png)

5. **Create a default epic and update the board filter query**

    - Go to the newly created backlog and create an epic named "Default"
    - On either the backlog or board page, click the ... button and click Board Settings
    - Under the "General" section click "Edit Filter Query"
    - Update the query to include a lookup for issues in your project's epics.
    - Click save

    ![screenshot 1568932989](https://user-images.githubusercontent.com/123595/65286628-4549cb80-db0e-11e9-88ea-f21730151493.png)
    ![screenshot 1568933000](https://user-images.githubusercontent.com/123595/65286629-45e26200-db0e-11e9-9083-b446574d81bc.png)
    ![screenshot 1568933065](https://user-images.githubusercontent.com/123595/65286630-45e26200-db0e-11e9-8d74-9179818d364e.png)

6. **Create your first ticket!**

    - Assign it to your Default epic.
    - Once a ticket is ready to be picked up, assign a user and move the project into their team project:
    
     ![screenshot 1568933669](https://user-images.githubusercontent.com/123595/65286903-3adc0180-db0f-11e9-8e6d-334411e1ab2c.png)
    ![screenshot 1568933678](https://user-images.githubusercontent.com/123595/65286904-3adc0180-db0f-11e9-9ef6-d680137deb92.png)

Because we've updated the project query to look for project issues OR issues within our project epics, you'll be able to track progress of a ticket within the original Task Force / Working Group project which increasing visibility of such tasks within product teams.

![screenshot 1568933757](https://user-images.githubusercontent.com/123595/65286905-3adc0180-db0f-11e9-963c-007924f32d14.png)
![screenshot 1568933794](https://user-images.githubusercontent.com/123595/65286906-3b749800-db0f-11e9-8192-28739b1f83ef.png)

**Comment by dblandin** [comment]
## Resolution

We've decided to document the process for creating a Jira Project intended to track work within a Task Force, Working Group, Practice, or similar non-product group. Groups will be able to manage a backlog, track status on a Kanban board, and move tickets to product projects while retaining visibility within the original project via epics.

## Level of Support

2: Positive feedback.

#### Additional Context:

We've decided not to implement a org-wide view of all non-product work taking place. This move to Jira and documenting a recommended project structure already offers consolidating benefits. We can look into creating an additional view later if still desired.

There were some ideas around tracking all non-product work within a single Jira project. This proved difficult to manage as each group will likely want their own backlog and instead of pushing against Jira norms and creating workarounds, we decided to lean into the project structure and recommend a separate project per group.

## Next Steps

- Move the documentation provided above into its own artsy/README page
- Migrate the On-Call and Design Systems working groups over to the new Jira structure
- Communicate these examples and work with other groups who are interested in also adopting the structure

#### Exceptions

None
